app/livechat/server/api/lib/inquiries.js

```diff
@@ -0,0 +1,64 @@
+import { hasPermissionAsync } from '../../../../authorization/server/functions/hasPermission';
+import { LivechatDepartmentAgents, LivechatDepartment, LivechatInquiry } from '../../../../models/server/raw';
+import { hasRoleAsync } from '../../../../authorization/server/functions/hasRole';
+
+export async function findInquiries({ userId, department, pagination: { offset, count, sort } }) {
+	if (!await hasPermissionAsync(userId, 'view-l-room')) {
+		throw new Error('error-not-authorized');
+	}
+
+	const options = {
+		limit: count,
+		sort: sort || { ts: -1 },
+		skip: offset,
+	};
+
+	if (department) {
+		const cursor = LivechatInquiry.find({ status: 'queued', department }, options);
+
+		const total = await cursor.count();
+
+		const inquiries = await cursor.toArray();
+
+		return {
+			inquiries,
+			count: inquiries.length,
+			offset,
+			total,
+		};
+	}
+
+	let departmentIds;
```

Comment:
Since the endpoint accepts a `department` field, I don't think you should add more validations here.
I mean, if this method is called on the `inquiries.all`, you need to rely only on the parameters the endpoint supports.
You want to add mandatory validations, you should create another endpoint for  this purpose, don't you think?
Like, `agent.inquiries`, or `queu.inquiries`, or something like this, then you could add those validations there and control this on the server side.